### PR TITLE
Add authorization gem

### PR DESCRIPTION
**Pull request by jestradadeveloper**
# Description
Add authorization gem in order to restrict what groups a user can edit and see

### CODE REVIEW

**Review by JuanequeX — APPROVED** [review]
LGMT

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,2 +1,7 @@
 class ApplicationController < ActionController::Base
+    
+
```
Comment (YilKanda) [inline app/controllers/application_controller.rb:3]: Delete one of the two spaces you left.

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -1,2 +1,7 @@
 class ApplicationController < ActionController::Base
+    
+    
+    rescue_from CanCan::AccessDenied do |exception|
+        redirect_to root_path
+    end    
 end
```
Comment (YilKanda) [inline app/controllers/application_controller.rb:7]: Check the indentation of your file.

File: app/controllers/groups_controller.rb [hunk 3]
```diff
@@ -1,8 +1,13 @@
 class GroupsController < ApplicationController
+  load_and_authorize_resource
   before_action :set_group, only: [:show, :edit, :update, :destroy]
   before_action :authenticate_user!, :except => [:show, :index]
   def index
-    @groups = Group.all
+    @groups = Group.joins(:participants).where(
+      'owner_id = ? OR participants.user_id = ?',
+      current_user.id,
+      current_user.id
```
Comment (YilKanda) [inline app/controllers/groups_controller.rb:9]: `current_user.id` must be repeated in its parameters?

File: app/models/ability.rb [hunk 4]
```diff
@@ -0,0 +1,11 @@
+# frozen_string_literal: true
+
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    can :manage, Group, owner_id: user.id
+	  can :read, Group, participating_users: { user_id: user.id }
+   
+  end
```
Comment (YilKanda) [inline app/models/ability.rb:10]: Check the indentation of your file.

**Review by YilKanda — COMMENTED** [review]
I recommend that you do a check on your code with Rubocop, since you have some details with the spaces and the indentation.

Regarding logic, I see that it looks good.

File: app/controllers/application_controller.rb (hunk 1 above)
Comment (jestradadeveloper) [reply app/controllers/application_controller.rb]: Thanks i will be more careful with the indentation and spaces issues. I already made this change.

File: app/controllers/application_controller.rb (hunk 2 above)
Comment (jestradadeveloper) [reply app/controllers/application_controller.rb]: Thanks i will be more careful with the indentation and spaces issues. I already made this change.

File: app/models/ability.rb (hunk 4 above)
Comment (jestradadeveloper) [reply app/models/ability.rb]: Thanks i will be more careful with the indentation and spaces issues. I already made this change.

**Review by YilKanda — APPROVED** [review]
LGTM!

File: app/controllers/groups_controller.rb (hunk 3 above)
Comment (jestradadeveloper) [reply app/controllers/groups_controller.rb]: i  did a refactor about this logic, now i only reference to current_user.id once in a scope method in Group model. Thanks for your CR

**Review by JuanequeX — APPROVED** [review]
LGTM

**Pull request merged**